vnext/Shared/HermesSamplingProfiler.h

```diff
@@ -0,0 +1,21 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+#pragma once
+
+#include <string>
+
+namespace Microsoft::ReactNative {
+
+class HermesSamplingProfiler final {
```

Comment:
does the profiler apply to all react native instances in the process, or is it specific to a particular instance? the statics make it seem like it's process-wide

Reply:
Yep.. It's process wide .. Enabling profiler starts sampling all the runtimes ..

Reply:
Does it have to be?  I doubt we'd want it to be process wide when trying to analyze a single SDX in office.

Reply:
@acoates-ms It's a restriction on the API, no fundamental restriction in Hermes .. Strictly speaking Hermes Samping profiler is not process wide .. Individual Jsi::Runtime can be enabled for profiling through the RuntimeConfig property sent to the makeHermesRuntime factory method .. Calling HermesRuntime::EnableSampleingProfiler starts the sampling thread which samples all the runtimes enabled for profiling .. We could change/Add the API to sample specific Runtime ..

Reply:
What do you think about adding "enableCPUSamplingProfiler" flag to devSettings which is defaulted to false .. Setting this flag to true will set the flag in the Hermes RuntimeConfig sent to makeHermesRuntime.. This way we can selectively sample runtimes without changing the current  Hermes API  ..